crates/bevy_ecs/src/query/access.rs

```diff
@@ -54,76 +62,115 @@ impl<T: SparseSetIndex> Access<T> {
         }
     }
 
-    /// Returns true if this `Access` contains a write access for the given index.
+    /// Returns `true` if this can exclusively access the element given by `index`.
     pub fn has_write(&self, index: T) -> bool {
-        self.writes.contains(index.sparse_set_index())
+        if self.writes_all {
+            true
+        } else {
+            self.writes.contains(index.sparse_set_index())
+        }
     }
 
-    /// Sets this `Access` to having read access for all indices.
-    pub fn read_all(&mut self) {
+    /// Sets this as having access to all indexed elements.
+    pub(crate) fn read_all(&mut self) {
         self.reads_all = true;
     }
 
-    /// Returns true if this `Access` has read access to all indices.
-    pub fn reads_all(&self) -> bool {
-        self.reads_all
+    /// Sets this as having exclusive access to all indexed elements.
+    pub(crate) fn write_all(&mut self) {
+        self.reads_all = true;
+        self.writes_all = true;
     }
```

Comment:
Arguments against making these private?

Reply:
Users attempting to write their own `World`-like APIs would need to hand-roll something? That's fine though. Private them plz.